mobile/src/main/java/ch/epfl/sdp/mobile/application/StoreDocuments.kt

```diff
@@ -46,3 +50,116 @@ fun ProfileDocument?.toProfile(currentUserUid: String?): Profile {
 fun ProfileDocument?.toProfile(currentUser: AuthenticationUser): Profile {
   return toProfile((currentUser as? AuthenticatedUser)?.uid)
 }
+
+/**
+ * A document which represents a game of chess between two users. All the game documents are stored
+ * in the `/games/` collection.
+ *
+ * @param moves
+ * @param whiteId
+ * @param blackId
+ */
+data class ChessDocument(
+    @DocumentId val uid: String? = null,
+    val moves: List<String>? = null,
+    val whiteId: String? = null,
+    val blackId: String? = null,
+)
+
+/**
+ * Deserializes and creates a [Match] from a [ChessDocument]
+ *
+ * For a null [ChessDocument], a default [Match] with
+ * null value and default [Game] positions will be created
+ *
+ * @return The deserialized [Match]
+ */
+fun ChessDocument?.deserialize(): Match {
```

Comment:
Great work on algebraic notation support. We should see how we can move this in the `application.chess.notation` package.